### PR TITLE
Allow daemonize in debug mode.

### PR DESCRIPTION
**Pull request by hjoliver**
Fixes a long-standing annoyance (for me anyway): `--no-detach` and `--debug` should be orthogonal.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
Need to make this change to get the failed test to pass:

```diff
diff --git a/tests/events/suite/suite.rc b/tests/events/suite/suite.rc
index f44da3b29..f26e633c7 100644
--- a/tests/events/suite/suite.rc
+++ b/tests/events/suite/suite.rc
@@ -9,7 +9,7 @@
         script = """
 cylc reg $REG $DEF
 echo "Sub-suite log file is: $PWD/$LOG"
-if cylc run --debug $REG > $LOG 2>&1; then 
+if cylc run --no-detach --debug $REG > $LOG 2>&1; then
     echo "ERROR: sub-suite did not abort as planned"
     exit 1
 else
```

**Comment by hjoliver** [comment]
Thanks - I guess I missed that one!

**Comment by matthewrmshin** [comment]
@oliver-sanders please sanity check.

File: lib/cylc/daemonize.py [hunk 1]
```diff
@@ -70,21 +70,27 @@ def daemonize(server):
...
-                    # LOG-PREIFX Suite starting: server=HOST:PORT, pid=PID
-                    # Otherwise, something has gone wrong, print the suite log
-                    # and exit with an error.
-                    log_line1 = open(log_fname).readline()
-                    if server.START_MESSAGE_PREFIX in log_line1:
-                        server_str, pid_str = log_line1.rsplit()[-2:]
-                        suite_pid = pid_str.rsplit("=", 1)[-1]
-                        suite_port = server_str.rsplit(":", 1)[-1]
-                    else:
+                    if len(open(log_fname).readlines()) < 2:
```
[the remark this answers is not in the document]
Comment (hjoliver) [reply lib/cylc/daemonize.py]: Good idea, will do that...

**Comment by hjoliver** [comment]
@oliver-sanders - comment addressed

**Pull request merged**